posthog/hogql/database/database.py

```diff
@@ -289,7 +289,7 @@ def create_hogql_database(
     for table in (
         DataWarehouseTable.objects.filter(team_id=team.pk)
         .exclude(deleted=True)
-        .select_related("credential", "external_data_source")
+        .prefetch_related("credential", "external_data_source")
```

Comment:
prefetch iirc loads the data and joins in python... if you try to select_related and it should be prefetch instead then you'll get an error

(but django orm constantly surprises me in a bad way so 🙈)

Reply:
original was in [posthog.tasks.tasks.update_cache_task](https://posthog.sentry.io/performance/summary/?project=1899813&referrer=performance-transaction-summary&transaction=posthog.tasks.tasks.update_cache_task&unselectedSeries=p100%28%29&unselectedSeries=avg%28%29)

this is in [/api/projects/{parent_lookup_team_id}/external_data_sources/](https://posthog.sentry.io/performance/summary/?project=1899813&referrer=performance-transaction-summary&transaction=%2Fapi%2Fprojects%2F%7Bparent_lookup_team_id%7D%2Fexternal_data_sources%2F&unselectedSeries=p100%28%29&unselectedSeries=avg%28%29)

Reply:
interestingly you can see the join happening in the trace

![Uploading Screenshot 2024-07-12 at 10.04.52.png…]()

Reply:
ah... it queries a second time...

![2024-07-12 10 07 20](https://github.com/user-attachments/assets/13e7dd9d-ea09-4d69-abfb-d335f72ee5e4)

and that second time doesn't select related...